### PR TITLE
Add Github precommit workflow

### PR DESCRIPTION
**Pull request by lunaticare**
Add Github workflow to check code formatting in pull requests.

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/3060"></a>

### CODE REVIEW

**Comment by lunaticare** [comment]
@Ailrun now that #3059 is merged, we should be able to enforce pre-commit hooks by default :)

**Comment by Ailrun** [comment]
Oh I see where that Brittany issue happened... It's because https://github.com/haskell/haskell-language-server/blob/347a7187f20242540926b9927c59a45b18798c67/.pre-commit-config.yaml#L27 starts with `.` (which I believe shouldn't)

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: ./.github/actions/setup-build
+      with:
+        ghc: "9.2.3"
+        os: ${{ runner.os }}
```
Comment (Ailrun) [inline .github/workflows/pre-commit.yml:17]: You should not shorten the project name for this, I believe. (Otherwise format can change)

**Review by Ailrun — COMMENTED** [review]
It looks like there was a concurrent PR without formatting (for Stan integration). Could you format the code as well?

File: .github/workflows/pre-commit.yml (hunk 1 above)
Comment (lunaticare) [reply .github/workflows/pre-commit.yml:17]: Fixed!

**Review by Ailrun — APPROVED** [review]
LGTM

**Pull request merged**